ncrx/libncrx.c

```diff
@@ -713,17 +717,96 @@ static void terminate_msg_and_dict(struct ncrx_msg *msg)
 		msg->text[msg->text_len] = '\0';
 		msg->dict_len = len - msg->text_len - 1;
 		msg->dict++;
+	} else {
+		msg->dict_len = 0;
 	}
 }
 
 struct ncrx_msg *ncrx_next_msg(struct ncrx *ncrx)
 {
-	struct ncrx_msg *msg = msg_list_pop(&ncrx->retired_list);
+	struct ncrx_msg *start_msg, *next_msg;
+	uint64_t last_seen_cont_seq = 0;
 
-	if (msg)
-		terminate_msg_and_dict(msg);
+	start_msg = msg_list_peek(&ncrx->retired_list);
+	if (!start_msg) {
+		return NULL;
+	}
 
-	return msg;
+	last_seen_cont_seq = start_msg->seq;
+
+	/* If strlen(start_msg->text) is 0, this was already part of a CONT
+	 * sequence we merged so we don't need to consider it for further merging
+	 */
+	if (!ncrx->p.cont_merging_enabled || !strlen(start_msg->text)) {
```

Comment:
Can we test `!*start_msg->text` so we avoid scanning the entire string in the normal message case?